RecoJets/JetAlgorithms/interface/HEPTopTaggerV2.h

```diff
@@ -0,0 +1,347 @@
+    _q_zcut = q_zcut; _q_dcut_fctr = q_dcut_fctr; _q_exp_min = q_exp_min; _q_exp_max = q_exp_max; _q_rigidity =  q_rigidity; _q_truncation_fctr =  q_truncation_fctr;
+  }
+  void set_qjets_rng(CLHEP::HepRandomEngine* engine){ _rnEngine = engine;}
+
+  //get information
+  bool is_maybe_top() const {return _is_maybe_top;}
+  bool is_masscut_passed() const {return _is_masscut_passed;}
+  bool is_minptcut_passed() const {return _is_ptmincut_passed;}
+  bool is_tagged() const {return (_is_masscut_passed && _is_ptmincut_passed);}
+
+  double delta_top() const {return _delta_top;}
+  double djsum() const {return _djsum;}
+  double pruned_mass() const {return _pruned_mass;}
+  double unfiltered_mass() const {return _unfiltered_mass;}
+
+  double f_rec();
+  const PseudoJet & t() const {return _top_candidate;}
+  const PseudoJet & b() const {return _top_subjets[0];}
+  const PseudoJet & W() const {return _W;}
+  const PseudoJet & W1() const {return _top_subjets[1];}
+  const PseudoJet & W2() const {return _top_subjets[2];}
+  const std::vector<PseudoJet> & top_subjets() const {return _top_subjets;}
+  const PseudoJet & j1() const {return _top_subs[0];}
+  const PseudoJet & j2() const {return _top_subs[1];}
+  const PseudoJet & j3() const {return _top_subs[2];}
+  const std::vector<PseudoJet> & top_hadrons() const {return _top_hadrons;}
+  const std::vector<PseudoJet> & hardparts() const {return _top_parts;}
+  const PseudoJet & fat_initial() {return _fat;}
+  
+  void get_setting() const;
+  void get_info() const;
+
+  double nsub(fastjet::PseudoJet jet, int order, fastjet::contrib::Njettiness::AxesMode axes = fastjet::contrib::Njettiness::kt_axes, double beta = 1., double R0 = 1.);
+  double q_weight() {return _qweight;}
+   
+private:
+  bool _do_qjets;
+
+  PseudoJet _jet;
+  PseudoJet _initial_jet;
+
+  double _mass_drop_threshold;
+  double _max_subjet_mass;
+
+  Mode _mode;
+  double _mtmass, _mwmass;
+  double _mtmin, _mtmax;
+  double _rmin, _rmax;
+  double _m23cut, _m13cutmin, _m13cutmax;
+  double _minpt_tag;
+  
+  unsigned _nfilt;
+  double _Rfilt;
+  JetAlgorithm _jet_algorithm_filter;
+  double _minpt_subjet;
+  
+  JetAlgorithm _jet_algorithm_recluster;
+
+  double _zcut;
+  double _rcut_factor;
+
+  double _q_zcut, _q_dcut_fctr, _q_exp_min, _q_exp_max, _q_rigidity, _q_truncation_fctr;
+  JetDefinition _qjet_def;
+  
+  PseudoJet _fat;
+
+  CLHEP::HepRandomEngine* _rnEngine;
+
+  bool _debug;
+  
+  bool _is_masscut_passed;
+  bool _is_ptmincut_passed;
+  bool _is_maybe_top;
+
+  double _delta_top;
+  double _djsum;
+
+  double _pruned_mass;
+  double _unfiltered_mass;
+
+  double _fw;
+  unsigned _parts_size;
+
+  PseudoJet _top_candidate;
+  PseudoJet _W;
+  std::vector<PseudoJet> _top_subs;
+  std::vector<PseudoJet> _top_subjets;
+  std::vector<PseudoJet> _top_hadrons;
+  std::vector<PseudoJet> _top_parts;
+
+  static bool _first_time;
```

Comment:
static code analyzer reports an issue with this variable
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14837/13448/llvm-analysis/
non-const statics should be avoided